chpl/chpl-service/src/main/java/gov/healthit/chpl/aspect/QuestionableActivityListener.java

```diff
@@ -58,16 +49,70 @@ public void setEnvironment(final Environment e) {
 
     /**
      * Any activity added with a reason would be handled here.
-     * @param concept - ActivityConcept
-     * @param objectId - Long
-     * @param activityDescription - String
-     * @param originalData - Object
-     * @param newData - Object
+     * @param concept
+     * @param objectId
+     * @param activityDescription
+     * @param originalData
+     * @param newData
+     * @param reason
      */
+    @After("execution(* gov.healthit.chpl.manager.impl.ActivityManagerImpl.addActivity(..)) && "
+            + "args(concept,objectId,activityDescription,originalData,newData,reason,..)")
+    public void checkQuestionableActivityWithReason(final ActivityConcept concept,
+            final Long objectId, String activityDescription, final Object originalData,
+            final Object newData, final String reason) {
+        LOGGER.info("Called QuestionableActivityAspect2.checkQuestionableActivityWithReason()");
+        if (originalData == null || newData == null
+                || !originalData.getClass().equals(newData.getClass())
+                || Util.getCurrentUser() == null) {
```

Comment:
Is there a way for there to be no current user? If so, does that mean it won't count as questionable activity? That seems off

Reply:
This is existing logic.  I removed this method thinking it was the cause of some failing tests.  Before I moved the methods to this class it existed in the QuestionableActivityAspect class.  After further research found that the problem existed in the QuestionableActivityManager.  I'm not 100 sure about this code, but since it had been working, I simply put it back in.